### ISSUE TITLE
axoupdater probably dumping weird tempdir garbage all over the place on windows

### ISSUE THREAD

**Issue by Gankra** [axodotdev/cargo-dist#1374]
Two things building up on my system:

PATH starts with this huge mess

```
C:\Users\ninte\AppData\Local\Temp\.tmpgnZRgs\bin;C:\Users\ninte\AppData\Local\Temp\.tmpwuPfF4\bin;C:\Users\ninte\AppData\Local\Temp\.tmpEzIVd6\bin;C:\Users\ninte\AppData\Local\Temp\.tmpIAVvIP\bin;C:\Users\ninte\AppData\Local\Temp\.tmp8odYwJ\bin;C:\Users\ninte\AppData\Local\Temp\.tmpxryyGS\bin;C:\Users\ninte\AppData\Local\Temp\.tmpr3cko9\bin............................
```

and .cargo is filling up with these:

![image](https://github.com/user-attachments/assets/5d724e2d-9612-4b30-8a07-9d8c74c6e4f5)

Seemingly the raw powershell installer doesn't have this problem, so I'm forced to conclude this is axoupdater's doing.

**Comment by Gankra** [comment]
clarification: this doesn't break anything but is disgusting if you ever notice it

**Comment by mistydemeo** [comment]
Just had a thought. Does a new one of those appear when you `cargo test` on axoupdater? The tests are probably leaking changes to your real environment, which we should work harder to disable.

It looks like the shell installer respects a `INSTALLER_NO_MODIFY_PATH` environment variable, but the PowerShell installer doesn't. It'd be worth unifying that behaviour.

Those temporary directories look like the result of binary relocation we do. Specifically, we [relocate the running binary](https://github.com/axodotdev/axoupdater/blob/main/axoupdater/src/lib.rs#L454-L477) to a temporary directory on the same drive so we can then replace it with the new binary at its original path. This directory is supposed to be deleted when the installation completes, but it looks like that must not be working.

**Comment by mistydemeo** [comment]
Just realized what must be happening. It's failing to delete the temporary directory because the running EXE is still in it, so it's leaving the directory and the old version of the EXE in place. This wasn't an issue before when we rooted the temporary directory in the system temp dir, since it would get cleaned up, but after https://github.com/axodotdev/axoupdater/pull/121 the temporary directories are surviving indefinitely.

I feel like I remembered somewhere seeing some kind of Windows `rm` command that acts after a lock is cleared...

**Comment by mitsuhiko** [comment]
I wonder if you could use [`self_replace`](https://github.com/mitsuhiko/self-replace/).  For very selfish reasons.  I use it in a few other places and it would be good to collect all these fixes for windows in one central crate :)

**Comment by mistydemeo** [comment]
I'm looking into `self_replace`. We have a usecase slightly outside of your expectations since we're performing the replacement in a subprocess, so we can't use the `self_replace` function directly, but I think we can work something out here!

**Comment by mistydemeo** [comment]
Fixed in axoupdater 0.7.3.
